navapbc/labs-decision-support-tool
# feat: chunk and store BEM pdf

## Ticket

https://navalabs.atlassian.net/browse/DST-342 

## Changes
> added regex to separate heading and contents from pdf
> set up file ingestion, parsing and db storage for BEM pdf


## Testing

1. add BEM policy manual to /app/documents directory
2. run `make ingest-policy-pdfs DATASET_ID=bridges-eligibility-manual BENEFIT_PROGRAM=multi-benefit BENEFIT_REGION=Michigan FILEPATH=/app/documents`
3. data should appear in DB (`SELECT * from document where dataset='bridges-eligibility-manual'`); get the document_id

<img width="1040" alt="Screenshot 2024-08-06 at 9 44 25 AM" src="https://github.com/user-attachments/assets/77454af6-1ca9-4a5a-ae2b-ab2758f7852d">
4. data should appear in DB (`SELECT * from chunk where document_id='{document_id}'`)

<img width="1005" alt="Screenshot 2024-08-06 at 9 41 53 AM" src="https://github.com/user-attachments/assets/b3a08656-5eb5-4dec-bc63-8bee23eae5c6">

## After deployment
[Load data to dev db](https://nava.slack.com/archives/C06DP498D1D/p1721397891286199?thread_ts=1721396512.937399&cid=C06DP498D1D)
